backend/danswer/server/manage/administrative.py

```diff
@@ -231,3 +235,56 @@ def update_token_budget_settings(
     # Store the settings in the dynamic config store
     get_dynamic_config_store().store(TOKEN_BUDGET_SETTINGS, settings_json)
     return {"message": "Token budget settings updated successfully."}
+
+
+@router.put("/admin/users")
+def bulk_invite_users(
+    _: User | None = Depends(current_admin_user),
+    emails: list[str] = Body(..., embed=True),
+) -> int:
+    all_emails = list(set(emails) | set(get_invited_users()))
+    return write_invited_users(all_emails)
+
+
+@router.patch("/admin/remove-invited-user")
+def remove_invited_user(
+    user_email: UserByEmail,
+    _: User | None = Depends(current_admin_user),
+) -> int:
+    users = get_invited_users()
+    remaining_users = [user for user in users if user != user_email.user_email]
+    return write_invited_users(remaining_users)
+
+
+@router.patch("/admin/block-user")
+def block_user(
+    user_email: UserByEmail,
+    _: User | None = Depends(current_admin_user),
+    db_session: Session = Depends(get_session),
+) -> None:
+    user_to_block = get_user_by_email(
+        email=user_email.user_email, db_session=db_session
+    )
+    if not user_to_block:
```

Comment:
may want to log warning if user is already blocked

Reply:
Better yet: users should probably not be allowed to themselves ^_^